### PR TITLE
chore(lint): use commitlint as part of lint

### PR DESCRIPTION
**Pull request by bufferoverflow**
This enforces the conventional commit guidelines described within [CONTRIBUTING.md](https://github.com/verdaccio/verdaccio/blob/master/CONTRIBUTING.md#git-commit-guidelines) as part of `npm run lint`.

**take care**, this would break the commits made via translation bot service, e.g. https://github.com/verdaccio/verdaccio/commit/3ea7f9a45fa6e64d8fa866996d328730c4fa3897

### CODE REVIEW

**Comment by juanpicado** [comment]
Interesting project @bufferoverflow . I'll test a bit and see the translation bot because that is out of my control.

**Comment by juanpicado** [comment]
@bufferoverflow I could not found any settings related to custom commit template on crowdin . Do you have any idea?

**Comment by bufferoverflow** [comment]
I just enabled the conventional config, but there are lot of other options https://github.com/marionebl/commitlint/blob/master/README.md#config

**Comment by bufferoverflow** [comment]
@juanpicado the exact config is [here](https://github.com/marionebl/commitlint/blob/master/%40commitlint/config-conventional/README.md#rules) and here's a copy of it:

```
[
  'build',
  'ci',
  'chore',
  'docs',
  'feat',
  'fix',
  'perf',
  'refactor',
  'revert',
  'style',
  'test'
]
```

**Comment by juanpicado** [comment]
thanks @bufferoverflow I'll check soon the tool :) I'm quite curious

**Comment by juanpicado** [comment]
Could you rebase? @bufferoverflow

**Comment by bufferoverflow** [comment]
introduced this `npm run changelog` for local verification. some ci systems like travis need special handling, I'll provide that later.

**Comment by juanpicado** [comment]
`npm run changelog` I'm not sure how it works. But, we use `standard-version` to generate it automatically.

**Comment by bufferoverflow** [comment]
I'll change that to `npm run commitlint`.

I use see `generate-changelog`
- https://github.com/bufferoverflow/verdaccio-gitlab/blob/master/package.json#L11
- https://github.com/bufferoverflow/verdaccio-gitlab#development

**Comment by juanpicado** [comment]
🤓 yeah I was inspecting your project

**Comment by juanpicado** [comment]
As you said ... crowding is not friendly :( 

```
> commitlint --from=master

⧗   input: Merge pull request #499 from verdaccio/renovate/css-loader-0.x
✔   found 0 problems, 0 warnings

⧗   input: Merge pull request #482 from verdaccio/renovate/pin-docker-digests
✔   found 0 problems, 0 warnings

⧗   input: Merge pull request #497 from verdaccio/fix-445
✔   found 0 problems, 0 warnings

⧗   input: test
✖   message may not be empty [subject-empty]
✖   type may not be empty [type-empty]
✖   found 2 problems, 0 warnings
⧗   input: chore: use commitlint
✔   found 0 problems, 0 warnings

⧗   input: chore(deps): update dependency css-loader to v0.28.9
✔   found 0 problems, 0 warnings

⧗   input: New Crowdin translations (#479)
✖   message may not be empty [subject-empty]
✖   type may not be empty [type-empty]
✖   found 2 problems, 0 warnings
⧗   input: chore(deps): update node to tag 9.2.0-alpine
✔   found 0 problems, 0 warnings

⧗   input: fix: IPv6 set up
✔   found 0 problems, 0 warnings

````

**Comment by bufferoverflow** [comment]
renamed command , added yarn.lock and hint to CONTRIBUTING.md

**Comment by bufferoverflow** [comment]
@juanpicado are the changes ok?

**Comment by juanpicado** [comment]
It is fine, I am just getting familiar with the tool :)

**Comment by bufferoverflow** [comment]
@juanpicado We can customize translation commits, see https://github.com/electron/electron-i18n/pull/200/files . I will add `commit_message: 'docs(website): new translations'` to `crowdin.yml`  and rebase this.

**Comment by juanpicado** [comment]
@bufferoverflow Don't we need to enable husky enabled as well?

**Comment by bufferoverflow** [comment]
yes, husky integration makes sense, I update this MR this evening and would appreciate if you could merge #516 already.

**Comment by juanpicado** [comment]
@bufferoverflow Merged, but, please only include on husky linting, do not include unit test since that will be handled by CI tools, just linting it's ok. thanks :)

**Comment by juanpicado** [comment]
Do not forget to rebase :)

File: package.json [hunk 1]
```diff
@@ -142,6 +146,7 @@
     "test:unit": "cross-env NODE_ENV=test BABEL_ENV=test jest '(/test/unit.*\\.spec|/test/webui/.*\\.spec)\\.js' --maxWorkers 2",
     "test:func": "cross-env NODE_ENV=test BABEL_ENV=test jest '(/test/functional.*\\.func)\\.js' --maxWorkers 2",
     "pre:ci": "npm run lint && npm run build:webui",
+    "commitmsg": "commitlint -e $GIT_PARAMS",
```
Comment (juanpicado) [inline package.json]: What that means?

File: package.json [hunk 2]
```diff
@@ -48,6 +48,9 @@
     "unix-crypt-td-js": "^1.0.0"
   },
   "devDependencies": {
+    "@commitlint/cli": "^6.0.2",
+    "@commitlint/config-conventional": "^6.0.2",
+    "@commitlint/travis-cli": "^6.0.2",
```
Comment (juanpicado) [inline package.json]: What it does `travis-cli`?

**Comment by juanpicado** [comment]
@bufferoverflow Have you noticed this is .... this is a trap.

I cannot merge because `master` branch is protected until CI approve the PR, but, `commitlint` is failing in CI. So, we have to find a way to bypass via configuration those failures until 

https://circleci.com/gh/verdaccio/verdaccio/1622?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```
⧗   input: New Crowdin translations (#479)
✖   message may not be empty [subject-empty]
✖   type may not be empty [type-empty]
✖   found 2 problems, 0 warnings
⧗   input: chore(deps): update node to tag 9.2.0-alpine
✔   found 0 problems, 0 warnings

⧗   input: fix: IPv6 set up
✔   found 0 problems, 0 warnings

⧗   input: fix #493 Update uplink documentation adding CA property
✖   message may not be empty [subject-empty]
✖   type may not be empty [type-empty]
✖   found 2 problems, 0 warnings
⧗   input: Update README.md
✖   message may not be empty [subject-empty]
✖   type may not be empty [type-empty]
✖   found 2 problems, 0 warnings
⧗   input: chore: update readme
✔   found 0 problems, 0 warnings
```

File: package.json (hunk 1 above)
Comment (bufferoverflow) [reply package.json]: This is called via commit hook, see https://github.com/marionebl/commitlint#getting-started or [commit-msg](https://git-scm.com/docs/githooks#_commit_msg) or [here](https://github.com/typicode/husky/blob/master/HOOKS.md) for the name matching between npm name and git.

File: package.json (hunk 2 above)
Comment (bufferoverflow) [reply package.json]: this provides [commitlint-travis](https://github.com/marionebl/commitlint/tree/master/%40commitlint/travis-cli), travis-ci does not a full git clone so special handling is required.

**Comment by bufferoverflow** [comment]
I did not figured out how circleci does handle git repos, I removed commitlint from circleci for now.

**Comment by bufferoverflow** [comment]
no change to circle.yml and it fails, at least travis is happy https://travis-ci.org/bufferoverflow/verdaccio/builds/331549165

**Comment by juanpicado** [comment]
@bufferoverflow no worries :) the tests were failing because of some network failure.

**Pull request merged**